modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -582,21 +578,11 @@ TX_ALREADY_FINISHED_ERR, format("Transaction is already finished [tx={}]", tx)
                     .parameters(params)
                     .timeZoneId(timeZoneId)
                     .defaultSchemaName(schemaName)
-                    .operationTime(tx.startTimestamp())
+                    .operationTime(operationTime)
+                    .txContext(txContext)
                     .build();
 
-            return executeParsedStatement(operationContext, result, txWrapper, null)
-                    .handle((executionResult, e) -> {
```

Comment:
this will be removed in https://issues.apache.org/jira/browse/IGNITE-21868 anyway